packages/terra-button/src/variants/_IconButton.jsx

```diff
@@ -0,0 +1,320 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import ThemeContext from 'terra-theme-context';
+import styles from '../Button.module.scss';
+import { ButtonVariants, ButtonTypes, IconTypes } from '../_constants';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * A icon to be displayed nested inline with the text when provided.
+   */
+  icon: PropTypes.element.isRequired,
+  /**
+   * specifies the icon type.
+   * refer Icon [Accessiblity Guide](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/icon/accessibility-guide) to know about types of icon.
+   */
+  iconType: PropTypes.oneOf([IconTypes.DECORATIVE, IconTypes.INFORMATIVE]).isRequired,
+  /**
+   * Whether or not the button should only display as an icon.
+   */
+  isIconOnly: PropTypes.bool,
```

Comment:
do we need this prop now ? as this file is specific to icon button

Reply:
yes we would still need this because icon-button supports both icon and icon with text variants.